### PR TITLE
optimize: remove variable deepest and reduce duplicate judgment

### PR DESCRIPTION
**Pull request by ruqinhu**
### Ⅰ. Describe what this PR did
1.(null instanceof Object) always be false
2. Depth has been judged by RollbackRule.getDepth(ex)

### CODE REVIEW

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java [hunk 1]
```diff
@@ -59,21 +59,16 @@ public void setRollbackRules(Set<RollbackRule> rollbackRules) {
...
         RollbackRule winner = null;
-        int deepest = Integer.MAX_VALUE;
-
         if (CollectionUtils.isNotEmpty(rollbackRules)) {
             winner = NoRollbackRule.DEFAULT_NO_ROLLBACK_RULE;
             for (RollbackRule rule : this.rollbackRules) {
                 int depth = rule.getDepth(ex);
-                if (depth >= 0 && depth < deepest) {
-                    deepest = depth;
+                if (depth >= 0) {
```
Comment (slievrly) [inline tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: rollbackRules contains AException and BException,  AException extend BException.  please valid this scenario.

Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: add class MyIllegalArgumentException extends IllegalArgumentException,and test is ok

Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: > rollbackRules contains AException and BException, AException extend BException. please valid this scenario.

add class MyIllegalArgumentException extends IllegalArgumentException,and test is ok

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java [hunk 2]
```diff
@@ -59,21 +59,16 @@ public void setRollbackRules(Set<RollbackRule> rollbackRules) {
...
                     winner = rule;
```
Comment (ruqinhu) [inline tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: rollbackRules is LinkedHashSet .  in this ,if i add break ,RollbackRule will be a Higher priority. 
but now, it just like as it was : Loop will execute to end ,and if RollbackRule and NoRollBackRule has same Exception, noRollBackRule will will take effect.

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 1 above)
Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: > rollbackRules contains AException and BException, AException extend BException. please valid this scenario.
add MyIllegalArgumentException extends IllegalArgumentException ,test is ok.
and I have a question below.

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 2 above)
Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: same Exception refers to the relationship of inheritance

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 1 above)
Comment (slievrly) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: It is always the smallest win. You have not compared their priority here. What is the point of the loop just assigning?

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 2 above)
Comment (slievrly) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: `if RollbackRule and NoRollBackRule has same Exception` Regardless of this scenario, users need to be responsible for their own actions. If you say both yes and no about a thing, it is impossible for the code to determine whether it is or not.

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 1 above)
Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: yes, return value is boolean ,Why compare priorities？if roolback and noroolback has same priority, i think below can add break for reducing cycle times...

Comment (ruqinhu) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: sorry, just thought if roolbackrule and noroolnack has father and son exception, smallest win will useful

**Comment by ruqinhu** [comment]
sorry, just thought if roolbackrule and noroolnack has father and son exception, smallest win will useful

**Comment by ruqinhu** [comment]
update test for showing smallest winner feature

**Pull request closed without merging**